src/core/caniuse.js

```diff
@@ -0,0 +1,206 @@
+      Can I Use this API?
+    </dt>`);
+  const placeholder = parent.appendChild(hyperHTML`
+    <dd class="caniuse-stats">fetching data from caniuse.com...</dd>`);
+
+  const cache = new IDBCache("respec-caniuse", ["caniuse"]);
+  await Promise.all([doc.respecIsReady, cache.ready]);
+
+  hyperHTML.bind(placeholder)`${{
+    any: canIUse(caniuse, cache),
+    placeholder: "fetching data from caniuse.com...",
+  }}`;
+}
+
+/**
+ * MUTATES `conf.caniuse` object to hold normalized configurarion
+ * @param {Object} conf   configuration settings
+ */
+function normalizeConf(conf) {
+  const DEFAULTS = {
+    maxAge: 24 * 60 * 60 * 1000, // 24 hours (in ms)
+    browsers: ["chrome", "firefox", "safari", "edge"],
+    versions: 4,
+  };
+
+  if (typeof conf.caniuse === "string") {
+    conf.caniuse = { feature: conf.caniuse, ...DEFAULTS };
+    return;
+  }
+  if (Array.isArray(conf.caniuse.browsers)) {
+    conf.caniuse.browsers = conf.caniuse.browsers
+      .map(b => b.toLowerCase())
+      .filter(isValidBrowser);
+  } else if (conf.caniuse.browsers !== "ALL") {
+    conf.caniuse.browsers = DEFAULTS.browsers;
+  }
+  if (conf.caniuse.maxAge === undefined) conf.caniuse.maxAge = DEFAULTS.maxAge;
+  if (!conf.caniuse.versions) conf.caniuse.versions = DEFAULTS.versions;
+
+  function isValidBrowser(browser) {
+    if (BROWSERS.has(browser)) {
+      return true;
+    }
+    pub("warn", `Ignoring invalid browser "\`${browser}\`" in \`conf.caniuse.browsers\``);
+    return false;
+  }
+}
+
+/**
+ * promises to get content for canIUse table
+ * @param {Object} conf              normalized respecConfig.caniuse
+ */
+async function canIUse(conf, cache) {
+  return new Promise(async (resolve) => {
+    // use data from cache data if valid and render
+    try {
+      const cachedStats = await cache.get(conf.feature);
+      if (cachedStats
+        && new Date() - new Date(cachedStats.cacheTime) < conf.maxAge) {
+        resolve(createTableHTML(conf.feature, cachedStats.stats, conf));
+      }
+    } catch (err) {
+      console.error(err);
+    }
+
+    // otherwise fetch new data, cache and render
+    const url = conf.apiURL
+      ? conf.apiURL.replace("{FEATURE}", conf.feature)
+      : `https://raw.githubusercontent.com/Fyrd/caniuse/master/features-json/${conf.feature}.json`;
+
+    try {
+      const json = await getJson(url);
+      cache.set(conf.feature, { stats: json.stats, cacheTime: new Date() })
+        .catch(err => console.error("Failed to cache caniuse data.", err));
+      resolve(createTableHTML(conf.feature, json.stats, conf));
+    } catch (err) {
+      const permalink = `http://caniuse.com/#feat=${conf.feature}`;
+      resolve(hyperHTML`
+        Error [core/caniuse]: ${err.message}.
+        <br>Please check directly on <a href="${permalink}">${permalink}</a>.`);
+    }
+  });
+}
+
+// TODO: replace with fetch in core/github ?
+async function getJson(url) {
+  const response = await window.fetch(url);
```

Comment:
So, essentially, `getJson()` should actually be `fetchAndCache(feature){}`. Essentially, the `canIUse()` function shouldn't care where the data comes from, just that it got data to render. 

So, we should do this here instead:

 1. When this module loads, create the cache instance. That can kick off initialization of IDB in the background. 
 1. When we are about to render, `fetchAndCache(feature)` (see below). 
 1.  We create the URL, which we can actually use as the unique identifier (this is good also if we want to use the browser's [Cache API](https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage), once it lands in Safari in a few months).
 1. await cache.ready.
 1. Check the cache for URL. If we have it, return the object.
 1. await, go to the network. 
 1. If error, deal with error. 
 1. await json.   
 1. Cache response.
 1. return json();

I think if we do the above here, the `canIUse()` function itself can go away. Then, this all boils down to:

```JS
hyperHTML.bind(placeholder)`${{
  any: getAndCacheJson(feature).then( json => createTableHTML(feature, json)),
  placeholder: "Fetching data from caniuse.com...",
}}`;
```

Reply:
I like this one! `fetchAndCache`!

Thoughts - What if we add some  more arguments to this function (or wrap it in some other function) so we can make use of a general `fetchAndCache` function that can be used in more places?
We can pass the cache and url to it as arguments.

Also, I feel like a function with "And" its name name should be refactored to two separate functions, although this might not be a case for doing that.

A little doubt - Use URL as the key for IDB?

Reply:
> Thoughts - What if we add some more arguments to this function (or wrap it in some other function) so we can make use of a general fetchAndCache function that can be used in more places?

I think we will definitely get there, but we basically want to evolve into that by road testing the code first. If we over generalize too early, we might end up over-engineering something.  

> Also, I feel like a function with "And" its name name should be refactored to two separate functions, although this might not be a case for doing that.

I don't have a strong opinion, tbh.   

> A little doubt - Use URL as the key for IDB?

I'd eventually like for us to just put things into the Cache API, which stores requests (which can be URLs) and responses. So, was trying to basically model it on that.